client/lib/themes/helpers.js

```diff
@@ -73,7 +73,19 @@ var ThemesHelpers = {
 	},
 
 	isPremium: function( theme ) {
-		return startsWith( theme.stylesheet, 'premium/' );
+		if ( ! theme ) {
+			return false;
+		}
+
+		if ( theme.stylesheet && startsWith( theme.stylesheet, 'premium/' ) ) {
+			return true;
+		}
+		// The /v1.1/sites/:site_id/themes/mine endpoint (which is used by the
+		// current-theme reducer, selector, and component) does not return a
+		// `stylesheet` attribute. However, it does return a `cost` field (which
+		// contains the correct price even if the user has already purchased that
+		// theme, or if they have an upgrade that includes all premium themes).
+		return theme.cost && theme.cost.number && theme.cost.number > 0;
```

Comment:
This is returning an integer instead of bool in certain instances, changing to:

```
return !! ( theme.cost && theme.cost.number && theme.cost.number > 0 );
```

Should fix it. :)